DogServiceExample/src/main/java/com/common/rest/DogService.java

```diff
@@ -0,0 +1,110 @@
+package com.common.rest;
+
+import com.common.json.JsonHelper;
+import com.common.app.Dog;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.Map;
+
+
+@Path("/")
+public class DogService {
+
+    private static Integer id = 0;
+    private static Map<Integer, Dog> dogs = new HashMap<Integer, Dog>();
+
+    static {
+        Dog dog0 = new Dog(id, "Rex", "08.11.2013", 40, 8, true);
+        dogs.put(dog0.getId(), dog0);
+
+        Dog dog1 = new Dog(id, "Bobik", "05.10.2011", 20, 3, true);
+        dogs.put(dog1.getId(), dog1);
+    }
+
+    public static void incrementDogId() {
+        id++;
+    }
+
+    @GET
+    @Path("dogs")
+    @Produces("application/json")
+    public Response printDogs() {
+        return Response.status(200).entity(JsonHelper.mapToJson(dogs)).build();
+    }
+
+    @GET
+    @Path("dog/{id}")
+    @Produces("application/json")
+    public Response printDog(@PathParam("id") String id) {
+        JsonElement jelement = new JsonParser().parse(JsonHelper.mapToJson(dogs));
+        JsonObject jobject = jelement.getAsJsonObject();
+        JsonObject result = jobject.getAsJsonObject(id);
```

Comment:
Ssame comments as in previous review, just get the dog from the map.
And don't forget to return 404 if no dog with such Id.